app/models/mixins/purging_mixin.rb

```diff
@@ -206,6 +206,8 @@ def purge_in_batches(conditions, window, total = 0, total_limit = nil)
           break if batch_ids.empty?
 
           current_window = batch_ids.size
+          _log.info("Purging Associated Records #{current_window} #{table_name.humanize}.")
+          purge_associated_records(batch_ids)
```

Comment:
Can we add a comment here something like

```suggestion
          # Purge the associated records before we purge the parent records to avoid leaving them orphaned
          purge_associated_records(batch_ids)
```

Reply:
Good idea.